lib/commanded/event_store.ex

```diff
@@ -3,48 +3,113 @@ defmodule OpentelemetryCommanded.EventStore do
 
   require OpenTelemetry.Tracer
 
-  alias OpenTelemetry.{Tracer, Span}
+  import OpentelemetryCommanded.Util
+
+  alias OpenTelemetry.Span
+
+  @tracer_id __MODULE__
 
   def setup do
-    :telemetry.attach_many(
-      {__MODULE__, :stop},
-      [
-        [:commanded, :event_store, :stream_forward, :stop],
-        [:commanded, :event_store, :append_to_stream, :stop]
-      ],
-      &__MODULE__.handle_stop/4,
-      []
-    )
+    ~w(
+      ack_event
+      adapter
+      append_to_stream
+      delete_snapshot
+      delete_subscription
+      read_snapshot
+      record_snapshot
+      stream_forward
+      stream_forward
+      stream_forward
+      subscribe
+      subscribe_to
+      subscribe_to
+      unsubscribe
+    )a
+    |> Enum.each(fn event ->
+      :telemetry.attach(
+        {__MODULE__, :start},
+        [:commanded, :event_store, event, :start],
+        &__MODULE__.handle_start/4,
+        []
+      )
 
-    :telemetry.attach_many(
-      {__MODULE__, :exception},
-      [
-        [:commanded, :event_store, :stream_forward, :exception],
-        [:commanded, :event_store, :append_to_stream, :exception]
-      ],
-      &__MODULE__.handle_stop/4,
-      []
-    )
+      :telemetry.attach(
+        {__MODULE__, :stop},
+        [:commanded, :event_store, event, :stop],
+        &__MODULE__.handle_stop/4,
+        []
+      )
+
+      :telemetry.attach(
+        {__MODULE__, :exception},
+        [:commanded, :event_store, event, :exception],
+        &__MODULE__.handle_exception/4,
+        []
+      )
+    end)
   end
 
-  def handle_stop([_, _, action, type], measurements, meta, _) do
-    end_time = :opentelemetry.timestamp()
-    start_time = end_time - measurements.duration
-    attributes = meta |> Map.take([:application, :stream_uuid]) |> Enum.to_list()
-    span_name = :"commanded.event_store.#{action}"
+  def handle_start([_, _, action, _type], _measurements, meta, _) do
+    event = meta.event
+
+    # TODO: do we want to trap this with helpful message about needing Middleware?
+    trace_headers = decode_headers(event.metadata["trace_ctx"])
+    :otel_propagator_text_map.extract(trace_headers)
+
+    attributes = [
+      "messaging.system": "commanded",
+      "messaging.protocol": "cqrs",
+      "messaging.destination_kind": "event_store",
+      # TODO does this need to change based on the event?  Or maybe `internal`?
+      "messaging.operation": "receive",
+      "messaging.message_id": event.causation_id,
+      "messaging.conversation_id": event.correlation_id,
+#      "messaging.destination": meta.handler_module,
```

Comment:
The above question about the span kind being "internal" is supported by the fact there is no known source or destination for this span.